cmd/podmanV2/images/search.go

```diff
@@ -0,0 +1,158 @@
+package images
+
+import (
+	"reflect"
+	"strings"
+
+	buildahcli "github.com/containers/buildah/pkg/cli"
+	"github.com/containers/buildah/pkg/formats"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/libpod/cmd/podmanV2/registry"
+	"github.com/containers/libpod/libpod/image"
```

Comment:
I don't think this is allowed.
@baude Correct?

Reply:
@mheon ?

Reply:
Concur, we're trying to avoid using libpod in podmanv2

Reply:
I would prefer waiting for the image-consolidation to be done (each line of code gets more expensive with every change) but sure, I can move the type somewhere else.

Reply:
I looked into it and don't think it's a good idea to break parts out of `libpod/image`. It could really be somewhere in `pkg/` and is not tied to `libpod`. Breaking parts out would IMO makes things worse.

Can we wait for the image consolidation?

Reply:
The issue, is a matter of size.  In that this will cause libpod to be pulled in swelling the size of the client.
But I will let @baude @jwhonce @mheon Decide on this.
One option would be, could we move libpod/image to pkg/image for the time being?

Reply:
Ah. In that case, the image package would pull in c/storage and a couple of other rather large packages. However, `libpod/image` is already pulled in via `podman load`.

Reply:
That's not really good, since separating Libpod and everything in the frontend is a core goal of this refactor.

Reply:
Alright. I'll work around it.

> That's not really good, since separating Libpod and everything in the frontend is a core goal of this refactor.

What is considered part of libpod? We are importing `libpod/define` in the v2 frontend. Is there a clear differentiator between what goes into `define` and what into `entities`?

I begin to see wisdom in Dan's proposal to enforce such rules in the CI.

Reply:
Define is Libpod-specific structures and constants (with a few exceptions - anything that goes in the DB, for example, needs to stay in Libpod). Anything struct or constant Libpod returns that doesn't fall under one of those exceptions will probably move to define/ eventually. The important point is that define is structs only, no code, and very minimal dependencies, so we can import it anywhere without much consequence.

As to what define is vs. entities, define is on the Libpod side of the libpod/Podman split, entities is on the Podman side. When we eventually split the repo, I expect entities to be moved into the new Podman repo.

Reply:
I worked around it by converting between `libpod/image` and the `entities` search result.

Regarding the desired clean separation between entities and libpod:

I had a closer look and fear that achieving the goal will require some further massaging until there's a clean cut. After some visual inspection, I ran `./dependencies/analyses/dependency-tree.sh` and found quite some (transitive) dependencies on `libpod`:

> github.com/containers/libpod/libpod, github.com/containers/libpod/libpod/define, github.com/containers/libpod/libpod/driver, github.com/containers/libpod/libpod/events, github.com/containers/libpod/libpod/image, github.com/containers/libpod/libpod/layers, github.com/containers/libpod/libpod/lock, github.com/containers/libpod/libpod/lock/file, github.com/containers/libpod/libpod/lock/shm, github.com/containers/libpod/libpod/logs, github.com/containers/libpod/libpod/logs/reversereader

There are many more low-level imports on buildah, docker, containerd, containers/storage etc.

I think that a clean separation requires a translation between low-level types and the high-level ones in `entities` (which then require copies of the low-level types).